src/spikeinterface/postprocessing/unit_locations.py

```diff
@@ -49,17 +49,43 @@ def __init__(self, sorting_analyzer):
         AnalyzerExtension.__init__(self, sorting_analyzer)
 
     def _set_params(self, method="monopolar_triangulation", **method_kwargs):
-        params = dict(method=method, method_kwargs=method_kwargs)
+        params = dict(method=method)
+        params.update(method_kwargs)
         return params
 
     def _select_extension_data(self, unit_ids):
         unit_inds = self.sorting_analyzer.sorting.ids_to_indices(unit_ids)
         new_unit_location = self.data["unit_locations"][unit_inds]
         return dict(unit_locations=new_unit_location)
 
+    def _merge_extension_data(
+        self, units_to_merge, new_unit_ids, new_sorting_analyzer, keep_mask=None, verbose=False, **job_kwargs
+    ):
+        arr = self.data["unit_locations"]
+        num_dims = arr.shape[1]
+        all_new_unit_ids = new_sorting_analyzer.unit_ids
+        counts = self.sorting_analyzer.sorting.count_num_spikes_per_unit()
+        new_unit_location = np.zeros((len(all_new_unit_ids), num_dims), dtype=arr.dtype)
+        for unit_ind, unit_id in enumerate(all_new_unit_ids):
+            if unit_id not in new_unit_ids:
+                keep_unit_index = self.sorting_analyzer.sorting.id_to_index(unit_id)
+                new_unit_location[unit_ind] = arr[keep_unit_index]
+            else:
+                id = np.flatnonzero(new_unit_ids == unit_id)[0]
+                unit_ids = units_to_merge[id]
+                keep_unit_indices = self.sorting_analyzer.sorting.ids_to_indices(unit_ids)
+                weights = np.zeros(len(unit_ids), dtype=np.float32)
+                for count, id in enumerate(unit_ids):
+                    weights[count] = counts[id]
+                weights /= weights.sum()
+                new_unit_location[unit_ind] = (arr[keep_unit_indices] * weights[:, np.newaxis]).sum(0)
```

Comment:
why not a clear recompute ? This is not always linear.

Reply:
What do you mean? We make a weighted sum of the former templates, based on the spike counts. You want to recompute everthing instead?

Reply:
Yes I think we should recompte the locations only for merges instead of make weithed average